### PR TITLE
fix: move static content import inside the specific method

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR fixes a runtime error due to a circular import. This caused XBlock not to be listed among the advanced components. These changes were tested in a remote environment with Redwood.

### How to Test
1. Install the XBlock with these changes in your Redwood environment.
2. Go to a unit in your course and select **Advanced**. You should see the Images Gallery component listed.

    image

### CODE REVIEW

**Comment by MaferMazu** [comment]
It works, but I have a question. Do you know why we need to put the imports inside the functions?

**Review by MaferMazu — APPROVED** [review]

**Comment by BryanttV** [comment]
Hi @MaferMazu, thanks for the review!

We must add the imports inside the functions to avoid circular imports when the CMS service is initialized.

<details>
<summary>Detailed error</summary>

```bash
cms-1  | 2024-09-02 14:23:37,055 WARNING 1 [xblock.plugin] [user None] [ip None] plugin.py:144 - Unable to load XBlock 'imagesgallery'
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2504, in resolve
cms-1  |     return functools.reduce(getattr, self.attrs, module)
cms-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  | AttributeError: partially initialized module 'imagesgallery.imagesgallery' has no attribute 'ImagesGalleryXBlock' (most likely due to a circular import)
cms-1  |
cms-1  | The above exception was the direct cause of the following exception:
cms-1  |
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 141, in load_classes
cms-1  |     yield (class_.name, cls._load_class_entry_point(class_))
cms-1  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 70, in _load_class_entry_point
cms-1  |     class_ = entry_point.load()
cms-1  |              ^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2496, in load
cms-1  |     return self.resolve()
cms-1  |            ^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2506, in resolve
cms-1  |     raise ImportError(str(exc)) from exc
cms-1  | ImportError: partially initialized module 'imagesgallery.imagesgallery' has no attribute 'ImagesGalleryXBlock' (most likely due to a circular import)
cms-1  | 2024-09-02 14:23:37,062 WARNING 1 [xblock.plugin] [user None] [ip None] plugin.py:144 - Unable to load XBlock 'imagesgallery'
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 141, in load_classes
cms-1  |     yield (class_.name, cls._load_class_entry_point(class_))
cms-1  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 70, in _load_class_entry_point
cms-1  |     class_ = entry_point.load()
cms-1  |              ^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2496, in load
cms-1  |     return self.resolve()
cms-1  |            ^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2502, in resolve
cms-1  |     module = __import__(self.module_name, fromlist=['__name__'], level=0)
cms-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/imagesgallery.py", line 20, in <module>
cms-1  |     from imagesgallery.edxapp_wrapper.contentstore import StaticContent, contentstore, update_course_run_asset
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/contentstore.py", line 40, in <module>
cms-1  |     StaticContent = get_static_content()
cms-1  |                     ^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/contentstore.py", line 15, in get_static_content
cms-1  |     backend = import_module(backend_function)
cms-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/opt/pyenv/versions/3.11.8/lib/python3.11/importlib/__init__.py", line 126, in import_module
cms-1  |     return _bootstrap._gcd_import(name[level:], package, level)
cms-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/backends/contentstore_r_v1.py", line 8, in <module>
cms-1  |     from cms.djangoapps.contentstore.views.assets import update_course_run_asset
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/__init__.py", line 6, in <module>
cms-1  |     from .course import *  # lint-amnesty, pylint: disable=redefined-builtin
cms-1  |     ^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/course.py", line 82, in <module>
cms-1  |     from ..tasks import rerun_course as rerun_course_task
cms-1  | ImportError: cannot import name 'rerun_course' from partially initialized module 'cms.djangoapps.contentstore.tasks' (most likely due to a circular import) (/openedx/edx-platform/cms/djangoapps/contentstore/tasks.py)
cms-1  | ImportError while importing <class 'ImportError'>
cms-1  | 2024-09-02 14:23:40,012 WARNING 30 [py.warnings] [user None] [ip None] warnings.py:110 - /openedx/venv/lib/python3.11/site-packages/pgpy/constants.py:5: DeprecationWarning: 'imghdr' is deprecated and slated for removal in Python 3.13
cms-1  |   import imghdr
cms-1  |
cms-1  | 2024-09-02 14:23:40,756 WARNING 30 [xblock.plugin] [user None] [ip None] plugin.py:144 - Unable to load XBlock 'imagesgallery'
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2504, in resolve
cms-1  |     return functools.reduce(getattr, self.attrs, module)
cms-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  | AttributeError: partially initialized module 'imagesgallery.imagesgallery' has no attribute 'ImagesGalleryXBlock' (most likely due to a circular import)
cms-1  |
cms-1  | The above exception was the direct cause of the following exception:
cms-1  |
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 141, in load_classes
cms-1  |     yield (class_.name, cls._load_class_entry_point(class_))
cms-1  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 70, in _load_class_entry_point
cms-1  |     class_ = entry_point.load()
cms-1  |              ^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2496, in load
cms-1  |     return self.resolve()
cms-1  |            ^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2506, in resolve
cms-1  |     raise ImportError(str(exc)) from exc
cms-1  | ImportError: partially initialized module 'imagesgallery.imagesgallery' has no attribute 'ImagesGalleryXBlock' (most likely due to a circular import)
cms-1  | 2024-09-02 14:23:40,759 WARNING 30 [xblock.plugin] [user None] [ip None] plugin.py:144 - Unable to load XBlock 'imagesgallery'
cms-1  | Traceback (most recent call last):
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 141, in load_classes
cms-1  |     yield (class_.name, cls._load_class_entry_point(class_))
cms-1  |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/xblock/plugin.py", line 70, in _load_class_entry_point
cms-1  |     class_ = entry_point.load()
cms-1  |              ^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2496, in load
cms-1  |     return self.resolve()
cms-1  |            ^^^^^^^^^^^^^^
cms-1  |   File "/openedx/venv/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2502, in resolve
cms-1  |     module = __import__(self.module_name, fromlist=['__name__'], level=0)
cms-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/imagesgallery.py", line 20, in <module>
cms-1  |     from imagesgallery.edxapp_wrapper.contentstore import StaticContent, contentstore, update_course_run_asset
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/contentstore.py", line 40, in <module>
cms-1  |     StaticContent = get_static_content()
cms-1  |                     ^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/contentstore.py", line 15, in get_static_content
cms-1  |     backend = import_module(backend_function)
cms-1  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/opt/pyenv/versions/3.11.8/lib/python3.11/importlib/__init__.py", line 126, in import_module
cms-1  |     return _bootstrap._gcd_import(name[level:], package, level)
cms-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/extra-deps/xblock-imagesgallery/imagesgallery/edxapp_wrapper/backends/contentstore_r_v1.py", line 8, in <module>
cms-1  |     from cms.djangoapps.contentstore.views.assets import update_course_run_asset
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/__init__.py", line 6, in <module>
cms-1  |     from .course import *  # lint-amnesty, pylint: disable=redefined-builtin
cms-1  |     ^^^^^^^^^^^^^^^^^^^^^
cms-1  |   File "/openedx/edx-platform/cms/djangoapps/contentstore/views/course.py", line 82, in <module>
cms-1  |     from ..tasks import rerun_course as rerun_course_task
cms-1  | ImportError: cannot import name 'rerun_course' from partially initialized module 'cms.djangoapps.contentstore.tasks' (most likely due to a circular import) (/openedx/edx-platform/cms/djangoapps/contentstore/tasks.py)
```
</details>

File: imagesgallery/imagesgallery.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 from xblock.reference.user_service import XBlockUser
```
Comment (mariajgrimaldi) [inline imagesgallery/imagesgallery.py:17]: Why don't we try removing this line instead? https://github.com/eduNEXT/xblock-imagesgallery/blob/main/imagesgallery/edxapp_wrapper/contentstore.py#L40

edit: I thought this was the error: https://3.basecamp.com/3966315/buckets/22136261/chats/3732788370@7766908568, not circular imports

Comment (BryanttV) [reply imagesgallery/imagesgallery.py:17]: It's a good idea! so we don't do the import inside the other methods.

Comment (BryanttV) [reply imagesgallery/imagesgallery.py:17]: Done: https://github.com/eduNEXT/xblock-imagesgallery/commit/6bd325ed6da7edf60096af457a9dc85824ce50c8

Comment (BryanttV) [reply imagesgallery/imagesgallery.py:17]: > edit: I thought this was the error: https://3.basecamp.com/3966315/buckets/22136261/chats/3732788370@7766908568, not circular imports

I'm not sure, I'm checking the CMS logs and I'm getting an error for circular import. https://github.com/eduNEXT/xblock-imagesgallery/pull/25#issuecomment-2324900188

Comment (mariajgrimaldi) [reply imagesgallery/imagesgallery.py:17]: Does moving the line work? Or do you keep getting the circular error exception?

Comment (BryanttV) [reply imagesgallery/imagesgallery.py:17]: Yes, moving the line directly to the method solves the error.

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**